### ISSUE TITLE
RNP generated EDDSA+X25519 private key fails with "bad passphrase"/"bad secret key" in GnuPG 2.2.19/libgcrypt 1.8.5

### ISSUE THREAD

**Issue by ronaldtse** [rnpgp/rnp#1383]
I do not know, if I should open up a new bug for this, because my problem is related to this.

I created a EDDSA + X25519 keypair in RNP and tried to use the private key with `gpg (GnuPG) 2.2.19/libgcrypt 1.8.5` which either fails with a `bad passphrase` or `bad secret key`.

I tried:
1. Created a key with `rnpkeys`
Then:
1. Exporting the private key with `rnpkeys`
2. Importing it to gpg. -> `Bad secret key`, but it got somehow imported
or
1. Use the keystore (pubring.gpg + secring.gpg) directly with gpg -> `bad passphrase` on export of private key, But displaying worked fine.

Should I open a dedicated but report?

_Originally posted by @func0der in https://github.com/rnpgp/rnp/issues/1018#issuecomment-761699333_

**Comment by ronaldtse** [comment]
_Originally from @func0der in https://github.com/rnpgp/rnp/issues/1018#issuecomment-761703750_

I tried to create a docker.sh file that could simulate the problem, but CentOS 7 has such an outdated libgcrypt version, that gpg does not support elliptic curves yet.
So you have to try it the manual way, sorry.

Commands ran:

```
export keydir=/tmp
rnpkeys --version -> rnp 0.12.0
rnpkeys --generate-key --expert --userid=foo@bar.de --homedir=${keydir} 
  -> Choose (22) EDDSA + X25519
  -> Passphrase: test123
rnpkeys --export-key --userid foo@bar.de --secret --homedir=${keydir}
(output in attached file `foo@bar.de.asc`)
```

[privkey+secring.gpg.zip](https://github.com/rnpgp/rnp/files/5825347/privkey%2Bsecring.gpg.zip)

**Comment by ronaldtse** [comment]
Ping @dewyatt @ni4 for triage.

**Comment by ni4** [comment]
This is a problem on the GnuPG side, I reported it here : https://dev.gnupg.org/T5114

**Comment by func0der** [comment]
@ni4 Neat. I could create a key with GnuPG and import it in and out of Thunbird (see bug mentioned above) and back into GnuPG without any problem. But I guess that the key'S first bit was just < 1, wans't it?
If I understand that correctly it was just (bad) luck, that I encountered the bug at all and furthermore in a non-persistent way?

**Comment by ni4** [comment]
Actually it's not a bad luck but flip of a coin. This should happen with 50% probability - EdDSA secret keys are generated as random 256 bits.

**Comment by ni4** [comment]
JFYI: This was fixed since GnuPG 2.2.24.

**Comment by ni4** [comment]
Okay, it was partially fixed for EdDSA secret key, however there is another part for Curve25519 subkey, see https://dev.gnupg.org/T5464
